MAINTAINERS.md

```diff
@@ -1,71 +1,137 @@
+    the release tag (from local and origin) and repeat the previous four
+    steps appropriately before requesting a follow-up review.
+  - If there are changes on the release preparation branch that should
+    *not* go to `master`, create a second branch
+    `release-<major>.<minor>-master` on `master` and cherry-pick all
+    relevant commits from the release branch preparation branch. Open a
+    pull request with that branch, go through review and push changes
+    back to the release preparation branch.
+- [ ] Go to the [release page][releases] to create a new release:
+    - [ ] Click on "Draft a new release".
+    - [ ] Name "`v<major>.<minor>`".
+    - [ ] Select the tag you created previously (it should be in
+          GitHub's dropdown list).
+    - [ ] Copy the [CHANGELOG][changelog] section for this release into
+          the description. (You don't need to provide the source `.zip`
+          and `.tar.gz` archives; GitHub does this automatically.)
+    - [ ] Release.
+- [ ] Merge `master` into the `release` branch and push to trigger
+      deployment.
+  - [ ] Check whether https://haskell.build/start is now the latest
+        [`start` script][start] (Netlify sometimes has problems).
+- [ ] Announce the new version on Twitter by asking someone with access.
+
+### Generating the PR List for the CHANGELOG
+
+The list of merged PRs can be fetched from:
+
+    https://github.com/tweag/rules_haskell/pulls?q=is:pr+base:master+merged:>YYYY-MM-DD
+
+Replacing `YYYY-MM-DD` by the date of the last release.
+
+If you have the GitHub CLI client, the following may be more convenient:
+
+```bash
+gh pr list -L 500 -B master -s merged \
+           --json number,mergedAt,title,body \
+| jq -r --argjson release "$(gh release view --json createdAt)" '
+     reverse | .[] | select(.mergedAt > $release.createdAt) |
+     ["# PR#\(.number): \(.title)", "*Merged: \(.mergedAt)*", "\(.body)\n"] |
+     join("\n\n")' \
+> PRs.md
+```
+
+**Note** The `-L 500` is an arbitrary "large number" limit of PRs to
+fetch, overriding the low default. As of writing, there's no way to set
+this to "unlimited"; adjust as necessary.
+
+### Generating the Archive Checksum
+
+The Git archive checksum can be calculated with:
+
+```bash
+git archive --format=tar.gz --prefix=rules-haskell-${REV}/ ${REV} | sha256sum
+```
+
+**Note** The trailing slash on the prefix is important; don't forget it.
+
+**Note** GitHub normalises release tags in the archive prefix to exclude
+the initial "v"; adjust as necessary.
+
+### Remove these PRs from Minor Releases
+
+- None
+
+<!-- Links -->
+[start]: ./start
+[versions]: ./haskell/private/versions.bzl
+[readme]: ./README.md
+[changelog]: ./CHANGELOG.md
+[usecases]: ./docs/haskell-use-cases.rst
+[releases]: https://github.com/tweag/rules_haskell/releases
+[checklist]: https://github.com/tweag/rules_haskell/blob/52f867daa2c3e67e1afff9a0460666ef9083e714/MAINTAINERS.md?plain=1#L29-L86
```

Comment:
Isn't this likely to go out of sync on future changes to this doc? Maybe it's good without the link to itself.

Reply:
It is, but I figured that this list would be relatively static beyond overhauls like this one. Still, I take your point that such changes can and will happen in the future and updating the link will likely be forgotten about.